kale/embed/video_se_i3d.py

```diff
@@ -0,0 +1,250 @@
+            model.Mixed_4b.add_module("SELayerTCc", SELayerC(512))
+            model.Mixed_4c.add_module("SELayerTCc", SELayerC(512))
+            model.Mixed_4d.add_module("SELayerTCc", SELayerC(512))
+            model.Mixed_4e.add_module("SELayerTCc", SELayerC(528))
+            model.Mixed_4f.add_module("SELayerTCc", SELayerC(832))
+            model.Mixed_5b.add_module("SELayerTCc", SELayerC(832))
+            model.Mixed_5c.add_module("SELayerTCc", SELayerC(1024))
+
+        self.model = model
+
+    def forward(self, x):
+        return self.model(x)
+
+
+class SEInceptionI3DFlow(nn.Module):
+    """
+    Add the several SELayers to I3D for optical flow input.
+    """
+
+    def __init__(self, num_channels, num_classes, attention):
+        super(SEInceptionI3DFlow, self).__init__()
+        model = InceptionI3d(in_channels=num_channels, num_classes=num_classes)
+        temporal_length = 16
+
+        # Add channel-wise SELayer
+        if attention == "SELayerC":
+            model.Mixed_3b.add_module("SELayerC", SELayerC(256))
+            model.Mixed_3c.add_module("SELayerC", SELayerC(480))
+            model.Mixed_4b.add_module("SELayerC", SELayerC(512))
+            model.Mixed_4c.add_module("SELayerC", SELayerC(512))
+            model.Mixed_4d.add_module("SELayerC", SELayerC(512))
+            model.Mixed_4e.add_module("SELayerC", SELayerC(528))
+            model.Mixed_4f.add_module("SELayerC", SELayerC(832))
+            model.Mixed_5b.add_module("SELayerC", SELayerC(832))
+            model.Mixed_5c.add_module("SELayerC", SELayerC(1024))
+
+        # Add temporal-wise SELayer
+        elif attention == "SELayerT":
+            model.Mixed_3b.add_module("SELayerT", SELayerT(temporal_length // 4))
+            model.Mixed_3c.add_module("SELayerT", SELayerT(temporal_length // 4))
+            model.Mixed_4b.add_module("SELayerT", SELayerT(temporal_length // 8))
+            model.Mixed_4c.add_module("SELayerT", SELayerT(temporal_length // 8))
+            model.Mixed_4d.add_module("SELayerT", SELayerT(temporal_length // 8))
+            model.Mixed_4e.add_module("SELayerT", SELayerT(temporal_length // 8))
+            model.Mixed_4f.add_module("SELayerT", SELayerT(temporal_length // 8))
+
+        self.model = model
+
+    def forward(self, x):
+        return self.model(x)
+
+
+def se_inception_i3d(name, num_channels, num_classes, attention, pretrained=False, progress=True, rgb=True):
+    """Get InceptionI3d module w/o SELayer and pretrained model."""
+    if rgb:
+        model = SEInceptionI3DRGB(num_channels, num_classes, attention)
+    else:
+        model = SEInceptionI3DFlow(num_channels, num_classes, attention)
+
+    if pretrained:
+        state_dict = load_state_dict_from_url(model_urls[name], progress=progress)
+
+        # delete logits.conv3d parameters due to different class number.
+        # state_dict.pop("logits.conv3d.weight")
+        # state_dict.pop("logits.conv3d.bias")
```

Comment:
Add a flag instead.

Reply:
Not easy to add a flag because need to change the output size of I3D in `video_i3d.py` file. Thus, I added more comments to introduce how to use it.

Reply:
That's fine. The comments are helpful.